bundle/edu.gemini.spdb.reports.collection/src/main/scala/edu/gemini/dbTools/ephemeris/TcsEphemerisExport.scala

```diff
@@ -0,0 +1,211 @@
+            }
+          }
+      }
+    } finally {
+      ctx.ungetService(odbRef)
+    }
+    log.info("Finish ephemeris lookup.")
+  }
+
+
+  private def formatCoords(coords: Coordinates): String = {
+    val ra  = Angle.formatHMS(coords.ra.toAngle, " ", 4)
+    val dec = Declination.formatDMS(coords.dec, " ", 3)
+
+    s"$ra $dec"
+  }
+
+  /** Writes the given ephemeris map to a String in the format expected by the
+    * TCS.
+    */
+  def formatEphemeris(em: EphemerisMap): String = {
+    val dfm = new SimpleDateFormat(DateFormatString)
+    dfm.setTimeZone(TimeZone.getTimeZone("UTC"))
+
+    em.toList.map { case (time, (coords, raTrack, decTrack)) =>
+      val timeS     = dfm.format(new Date(time))
+      val jdS       = f"${new JulianDate(time).toDouble}%.9f"
+      val coordsS   = formatCoords(coords)
+      val raTrackS  = f"$raTrack%9.5f"
+      val decTrackS = f"$decTrack%9.5f"
+      s" $timeS $jdS     $coordsS $raTrackS $decTrackS"
+    }.mkString("$$SOE\n", "\n", "\n$$EOE\n")
+  }
+
+  // Need Order to use HorizonsDesignation as a ==>> key
+  implicit val OrderHorizonsDesignation: Order[HorizonsDesignation] =
+    Order.orderBy(_.toString)
+}
+
+class TcsEphemerisExport(dir: File, night: Night, site: Site, odb: IDBDatabaseService, user: java.util.Set[Principal]) {
+  import TcsEphemerisExport._
+
+  val start = new Date(night.getStartTime)
+  val end   = new Date(night.getEndTime)
+
+  val lookupNonSiderealObservations: TryExport[HorizonsDesignation ==>> Set[String]] =
+    TryExport.fromTryCatch(OdbError) {
+      val ns = NonSiderealObservationFunctor.query(odb, user)
+      (==>>.empty[HorizonsDesignation, Set[String]]/:ns) { (m, n) =>
+        m.updateAppend(n.hid, Set(n.targetName))
+      }
+    }
+
+  val export: TryExport[HorizonsDesignation ==>> (ExportError \/ File)] =
+    lookupNonSiderealObservations >>= exportAll
+
+  private def exportAll(nos: HorizonsDesignation ==>> Set[String]): TryExport[HorizonsDesignation ==>> (ExportError \/ File)] =
+    EitherT(nos.mapOptionWithKey { (hid, names) =>
+      // There can be multiple names for the same horizons id.  We'll just pick
+      // one at random for now ...
+      names.headOption.map { exportOne(dir, hid, _) }
+    }.traverse(_.run).map(_.right[ExportError]))
+
+  def exportOne(dir: File, hid: HorizonsDesignation, name: String): TryExport[File] =
+    for {
+      em <- lookupEphemeris(hid)
+      f  <- writeEphemeris(hid, name, em)
+    } yield f
+
+  def lookupEphemeris(hid: HorizonsDesignation): TryExport[EphemerisMap] =
+    HorizonsService2.lookupEphemerisE[EphemerisElement](hid, site, start, end, ElementCount) { (ee: EphemerisEntry) =>
+      ee.coords.map((_, ee.getRATrack, ee.getDecTrack))
+    }.leftMap(e => HorizonsError(hid, e): ExportError)
+
+  def writeEphemeris(hid: HorizonsDesignation, name: String, em: EphemerisMap): TryExport[File] = {
```

Comment:
I didn't mean random in the strict sense of random numbers.  I meant that the "name" could contain anything since people type it in to the target name field.  I would feel better if the target name were sanitized before being included in the filename.

As I understand it, the target name is used as the ephemeris file name currently, and somehow the name is taken from there and written into the FITS header as the value the `OBJECT` key.